### DISCUSSION TITLE
ltex-ls hiding equations and performance

### DISCUSSION THREAD

**Discussion by davidcortesortuno** [LazyVim/LazyVim#1182] [General]
Hi, I'm using LazyVim and Mason with `ltex-ls` and I noticed that when moving the cursor out of a paragraph, equations and also words in bold/italics are rendered differently. Especially equations look wrong. Is this a LazyVim issue or something in the language server?

![Screenshot from 2023-07-21 16-59-57](https://github.com/LazyVim/LazyVim/assets/12200541/3ebc7111-2a01-4f45-8eec-74bd96d8b713)

![Screenshot from 2023-07-21 17-00-04](https://github.com/LazyVim/LazyVim/assets/12200541/9c591eb3-3aa0-4b20-83a7-14fc2d27e401)

I'm also editing a long latex file, but performance is very slow so I can barely type. When I type I see diagnostics running all the time. Is there a way to tune this or improve performance? I'm using Wezterm but the same happens in Gnome's console

![Screenshot from 2023-07-21 17-00-49](https://github.com/LazyVim/LazyVim/assets/12200541/8f421087-a1e4-46f0-8ecd-fc2c0f5a67fc)

**Comment by pkazmier** [comment]
I can’t speak to your other issues, but the part about text being displayed differently when you are on the line in question sounds like `conceallevel`. https://neovim.io/doc/user/options.html#'conceallevel'

**Reply by folke** [reply]
You could also try the tex extra. It sets conceallevel to 2 and uses vimtex for rendering instead of treesitter

**Reply by davidcortesortuno** [reply]
Aha, ok, thanks, I'll try those options :)

**Comment by gwarf** [comment]
For the continuous `ltex-ls` checks, you may try to use the ltex server options `checkFrequency`: https://valentjn.github.io/ltex/settings.html#ltexcheckfrequency.
By default it should be on edit and somewhat perpetual, but you can make them on save.
That said, still in my setup, while the settings apparently helped compared to using `checkFrequency = "edit"`, I see checks happening more often that I would like and it is sometimes delaying the Code Actions pop up, which is still a bit of a pain. But I didn't investigate more to understand what is happening.

I'm currently using something like this

```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        ltex = {
          settings = {
            -- https://valentjn.github.io/ltex/settings.html
            ltex = {
              -- XXX: unwanted checks are still occurring, often delaying CodeActions
              checkFrequency = "save",
              language = "en-GB",
              additionalRules = {
                enablePickyRules = true,
                motherTongue = "fr",
              },
            },
          },
        },
     },
  },
},
```

**Reply by jashmac** [reply]
I do not know if this helps in your case, but it is possible to disable the 'checking document' diagnostic notifications for ltex only, keeping other lsp servers unaffected. See my answer [here](https://github.com/folke/noice.nvim/issues/166#issuecomment-2467068148)

**Reply by davidcortesortuno** [reply]
Awesome, thanks!
